src/vm/comsynchronizable.cpp

```diff
@@ -1537,15 +1537,58 @@ void QCALLTYPE ThreadNative::nativeInitCultureAccessors()
     END_QCALL;
 }
 
+typedef HRESULT(WINAPI *pfnSetThreadDescription)(HANDLE hThread, PCWSTR lpThreadDescription);
+extern pfnSetThreadDescription g_pfnSetThreadDescription;
+
+HRESULT WINAPI InitializeSetThreadDescription(HANDLE hThread, PCWSTR lpThreadDescription)
+{
+    pfnSetThreadDescription func = NULL;
+
+#ifndef FEATURE_PAL
+    HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+    if (hKernel32 != NULL)
+    {
+        func = (pfnSetThreadDescription)GetProcAddress(hKernel32, "SetThreadDescription");
+    }
+
+    if (func != NULL) // method is only available with Windows 10 Creators Update
+    {
+        g_pfnSetThreadDescription = func;
+        return func(hThread, lpThreadDescription);
+    }
```

Comment:
LoadLibrary/GetProcAddress every time on pre-Windows 10 Creators Update. The function should be set to a dummy one after it failed once.